### DISCUSSION TITLE
Album Folder Navigation Bug

### DISCUSSION THREAD

**Discussion by R101G** [rern/rAudio#1243] [Fixed Issues]
@rern 
If I click on an album folder to view the tracks and then click on the back arrow, it displays the two top level source folders (NAS and USB), not the previous level's list of albums.

rAudio 20230218
6.1.11-1 aarch64
Raspberry Pi 3B Rev 1.2
BCM2837 • 957 MiB
4 x Cortex-A53 @ 1.20 GHz

**Comment by ceejayemm** [comment]
I can only reproduce this if I list Albums from Library/NAS, list tracks and then back arrow.  This takes me back to NAS and the contents of my Playlist folder (?) so not quite the same as @R101G 

If use Library/Albums, select album to show tracks then Back Arrow takes me back to Library as it should.

Chris

**Comment by R101G** [comment]
To clarify, I am selecting Library-USB-USBStick-Artist, then clicking to the right of the album icon to show the track listing. When I click the back arrow, I get:

NAS
USB

**Comment by klassikmann** [comment]
Same here.

**Comment by rern** [comment]
Please try the UPDATE branch.

**Reply by klassikmann** [reply]
@rern 
UPDATE seems to solve the problem.
Thank you

**Comment by ceejayemm** [comment]
@rern 
Thanks, yes the Update branch fixes this issue for me.

Chris

**Comment by R101G** [comment]
@rern 
Thanks. All back to normal now.

**Comment by clriis** [comment]
What is meant by "UPDATE branch?

I am running version "20230218" and facing the exact same problem

Claus

**Reply by ceejayemm** [reply]
Go to the Addons / rAudio Update section as normal then press and hold (aka long hold) the Update button and the Branch / Release should change to UPDATE then apply as normal.

Chris

**Comment by klassikmann** [comment]
@clriis 
But be careful, the UPDATE branch changed since rern's notice. UPDATE is his development branch and he made several enhancements since yesterday early mornig.
Advice: First make a backup (Settings menu), then try UPDATE. If new errors occur restore the backup and wait for the next release.

**Reply by clriis** [reply]
@ceejayemm and @klassikmann 
Thanks alot for the swift reply and explanation. 

I think I'll hold till the next release.

**Reply by Retrofreak** [reply]
I couldn't wait, so I just tried: Works! Thanks for explaining the "Update branch" feature ;-)  
Somehow navigation is much slower than before. but at least you end up back in the right place.

**Comment by klassikmann** [comment]
@clriis 
Release 20230224 containing the fix is available.
